File: src/common/common/main/java/settingdust/mod_sets/ModSets.java
package settingdust.mod_sets;

import net.minecraft.resources.ResourceLocation;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import settingdust.mod_sets.util.MinecraftAdapter;
import settingdust.mod_sets.util.SavingData;

public class ModSets {
    public static final String ID = "mod_sets";

    public static final Logger LOGGER = LoggerFactory.getLogger("ModSets");

    public static void load() {
        SavingData.invoker.load();
    }

    public static void save() {
        SavingData.invoker.save();
    }

    public static ResourceLocation id(String path) {
        return MinecraftAdapter.get().id(ID, path);
    }
}

File: src/common/main/kotlin/settingdust/mod_sets/ModSets.kt
package settingdust.mod_sets

import kotlinx.serialization.json.Json
import org.apache.logging.log4j.LogManager
import settingdust.mod_sets.data.SavingData
import settingdust.mod_sets.util.MinecraftAdapter

object ModSets {
    const val ID = "mod_sets"

    val LOGGER = LogManager.getLogger()

    val configJson = Json {
        encodeDefaults = true
        ignoreUnknownKeys = true
        prettyPrint = true
    }

    init {
        reload()
    }

    fun id(path: String) = MinecraftAdapter.id(ID, path)

    fun reload() {
        for (clazz in SavingData::class.sealedSubclasses) {
            clazz.objectInstance?.reload()
        }
    }

    fun save() {
        for (clazz in SavingData::class.sealedSubclasses) {
            clazz.objectInstance?.save()
        }
    }
}
